src/alignment.js

```diff
@@ -82,7 +82,7 @@ var alignWithDOM = function(nodeName, key, statics) {
   if (currentNode && matches(currentNode, nodeName, key)) {
     matchingNode = currentNode;
   } else {
-    var existingNode = key && getChild(parent, key);
+    var existingNode = /** @type {Element} */(key && getChild(parent, key));
```

Comment:
It doesn't seem like we should check for key both inside and outside getChild. Can we just check inside getChild?